modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSourceTestCase.java

```diff
@@ -589,69 +594,71 @@ public void getRetryPolicyExhaustedAndConnectionErrorsAreComputed() throws Excep
 
   @Test
   public void sourceInitializedLogMessage() throws Exception {
-    ArrayList<String> debugMessages = new ArrayList<>();
-    Logger oldLogger = setLogger(messageSource, LOGGER_FIELD_NAME, createMockLogger(debugMessages, DEBUG));
+    logger.resetLogs();
+    logger.setLevel(DEBUG);
     messageSource.initialise();
     if (primaryNodeOnly) {
-      verifyLogMessage(debugMessages,
+      verifyLogMessage(logger.getMessages(),
                        "Message source 'source' on flow 'appleFlow' running on the primary node is initializing. Note that this Message source must run on the primary node only.");
     } else {
-      verifyLogMessage(debugMessages,
+      verifyLogMessage(logger.getMessages(),
                        "Message source 'source' on flow 'appleFlow' is initializing. This is the primary node of the cluster.");
     }
-    setLogger(messageSource, LOGGER_FIELD_NAME, oldLogger);
+    logger.resetLevel();
```

Comment:
put this in an after of the test so that it will always be invoked even if an exception happens.